### ISSUE TITLE
NullPointer from SearchHandlerAppearanceTracker on iOS

### ISSUE THREAD

**Issue by Larhei** [dotnet/maui#19504]
### Description

Im getting a null pointer on iOS when navigating back to a page containing a Shell.Searchhandler.
CallStack is
`Unhandled managed exception: Object reference not set to an instance of an object. (System.NullReferenceException)
   at Microsoft.Maui.Platform.ColorExtensions.ToPlatform(Color color)
   at Microsoft.Maui.Controls.Platform.Compatibility.SearchHandlerAppearanceTracker.UpdateClearPlaceholderIconColor(Color targetColor)
   at Microsoft.Maui.Controls.Platform.Compatibility.SearchHandlerAppearanceTracker.UpdateTextColor(UITextField textField)
   at Microsoft.Maui.Controls.Platform.Compatibility.SearchHandlerAppearanceTracker.UpdateSearchBarColors()
   at Microsoft.Maui.Controls.Platform.Compatibility.ShellPageRendererTracker.SetAppeared()
   at Microsoft.Maui.Controls.Platform.Compatibility.ShellPageRendererTracker.PageAppearing(Object sender, EventArgs e)
   at Microsoft.Maui.Controls.Page.SendAppearing()
   at Microsoft.Maui.Controls.Shell.SendPageAppearing(Page page)
   at Microsoft.Maui.Controls.ShellContent.SendPageAppearing(Page page)
   at Microsoft.Maui.Controls.ShellContent.SendAppearing()
   at Microsoft.Maui.Controls.ShellSection.PresentedPageAppearing()
   at Microsoft.Maui.Controls.ShellSection.OnPopAsync(Boolean animated)`

Well I guess this is the reason for the error
 [Line](https://github.com/dotnet/maui/blob/08b71495293eaa70a618f1b0f24ca87046e7f787/src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/SearchHandlerAppearanceTracker.cs#L265C4-L265C82)

It should be
`uiButton.TintColor = targetColor?.ToPlatform() ?? _defaultPlaceholderTintColor;`

Also there is an other interesting code segments.
[Here](https://github.com/dotnet/maui/blob/08b71495293eaa70a618f1b0f24ca87046e7f787/src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/SearchHandlerAppearanceTracker.cs#L167)

In short. All calls to ToPlatform() for color should be xxx?.ToPlatform() in case there is no null check ;-)

### Version with bug

8.0.3

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS

### Did you find any workaround?

No
